text/0027-extensions.md

```diff
@@ -0,0 +1,299 @@
+# Project title
+- Start Date: 2020/08/20
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/27
+
+## Decision, including impact on distributions
+
+We'll use an extension system that supports
+1. discovery—that components know where to "plug themselves in" to the UI
+2. configurability—that implementers can change which components are plugged in where
+    using config files
+
+We'll build an UI in a new module `@openmrs/esm-implementer-tools`
+for managing that configuration, which will assist implementers
+in changing the config to add, disable, or reorder extensions.
+
+All servers will have a config file `config.json` in the `frontends` directory
+of the application data directory. This file can be written to by the server,
+and managed through the UI.
+
+## Definition
+
+An **Extension** is a component with a unique *name* and a *type*.
+
+An **Extension Point** is a place in the DOM where extensions can be attached.
```

Comment:
@jonathandick @rbuisson @ibacher Do you have thoughts about the naming?

Reply:
I'm pretty agnostic on this. "Extension point" has some history in the OpenMRS community, not just in the AppFramework, but it's also the name for the equivalent concept in the back-end module implementation. That said, "extension slot" does invoke the idea of a "space" rather than just, e.g., a join point.

Reply:
slot sounds good to me. would not try to match ref app which already seems to be understood by view

Reply:
(I should mention that the backend "extension points" are not, AFAIK, widely used.